### ISSUE TITLE
LEM labs don't use our F5 scripts

### ISSUE THREAD

**Issue by prometheanfire** [rcbops/rpc-openstack#990]
##### What version (commit) do you encounter this issue on?
All

##### What type of issue is this (bug, feature, documentation, regression)?
bug/regression

##### What is the expected behavior?
For the labs to use our F5 scripts

##### What is actually happening?
They have their own setup

##### How can we reproduce this issue?
Try running liberty without the fixes provided by #965 and using our F5 scripts

##### How can we verify your issue is addressed?
The labs are using our F5 scripts or some other way to keep in sync.

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?
No, this is more of a QE testing bug

##### What is your evaluation on the importance of this issue, and why?
unkn

----------------------------------------------------------------------------------------------------------------------------
###### For each pull request, update the branch line to reference it.
- [ ] master
- [ ] liberty-12.2
- [ ] kilo

##### Upgrade impact. Consider the impact on moving between major, minor and patch releases.
- [ ] This change does not impact upgrades
- [ ] This change addresses a potential upgrade impact
- [ ] The upgrade impact is being tracked by another issue.

The issue tracking the upgrade impact is:

##### Testing impact
- [ ] An issue/pull-request has been create or exists to track the testing gap, and is referenced below
- [ ] This change includes additional testing
- [ ] This change is tested by the existing setup

Additional testing links:

##### Release note
- [ ] This change includes a release note.
- [ ] This change does not require a release note.

##### Does your change require documentation?
- [ ] Yes, and I submitted an issue to [Docs issues](https://github.com/rackerlabs/docs-rpc/issues "Docs issues")
- [ ] Yes, and I contributed a documentation patch to [docs-rpc](https://github.com/rackerlabs/docs-rpc "docs-rpc")
- [ ] No documentation required for this change

Additional documentation links (please add links if you have ticked yes):

**Comment by prometheanfire** [comment]
@rcbops/rax-qa ^ @Mudpuppy12 @valw 

All of liberty were missing the fixes provided by #965 because of a mismatch of what we have vs what's run in LEM.

**Comment by sigmavirus24** [comment]
@rcbops/rax-qa @Mudpuppy12 @valw ping

**Comment by Mudpuppy12** [comment]
Actually it does use the f5_config,py script. It take the output of that script and uploads it into the F5 via REST api. Support / RPC just hands the output to netsec to configure the F5.

I think there would be a minor change to the script to make it work for both environments, and can be updated kept in rpc_openstack.

The biggest difference I can think of, is the LAB environments put the Configs in LAB_NAME partition on the F5. RPC uses default "RPC" partition.

Something like this:

if os.environ.has_key('LAB_NAME'):
    LAB_NAME = os.environ['LAB_NAME'].lower()
else:
      LAB_NAME='RPC'
      PREFIX_NAME='RPC'

PART = LAB_NAME.upper()
PREFIX_NAME = LAB_NAME.upper()

Might be needed to bring both scripts aligned. Let me take some time and make sure this was the only deviation between the lem f5, and support.

- Dennis

**Comment by Mudpuppy12** [comment]
denn8098@MN80JQG3QN:/tmp$ diff f5-config.py ~/PycharmProjects/rpc_lem/lem-aide/user-tools/f5_config.py 
19c19
< import os
---
> import os,sys
22a23,38
> if os.environ.has_key('LAB_DATA_DIR'):
>    # If we have LAD_DATA_DIR set, we are operating inside a lab environment
>    LAB_DATA_DIR = os.environ['LAB_DATA_DIR']
> else:
>        print "Incorrect LEM install missing LAB_DATA_DIR"
>        sys.exit(1)
> 
> 
> if os.environ.has_key('LAB_NAME'):
>     LAB_NAME = os.environ['LAB_NAME'].lower()
> else:
>       print "Incorrect LEM install missing LAB_NAME"
>       sys.exit(1)
> 
> PART = LAB_NAME.upper()
> PREFIX_NAME = LAB_NAME.upper()
24,25d39
< PART = 'RPC'
< PREFIX_NAME = 'RPC'
29a44
> 
41a57,59
> 
> ## TODO: Configure monitoring user
> 
54c72
<     r' destination *:80 recv "302 Found" send "HEAD / HTTP/1.1\r\nHost:'
---
>     r' destination *:80 recv "200 OK" send "HEAD / HTTP/1.1\r\nHost:'
60c78
<     r' https destination *:443 recv "302 FOUND" send "HEAD / HTTP/1.1\r\nHost:'
---
>     r' https destination *:443 recv "200 OK" send "HEAD / HTTP/1.1\r\nHost:'
255c273
<     'nova_console': {
---
>     'nova_spice_console': {
454c472
<         default=None
---
>         default=LAB_NAME.lower() + ".rpc.rackspace.com"
544,546c562,564
<         'run util bash',
<         'tmsh create ltm rule /' + PART + '/' + PREFIX_NAME + '_DISCARD_ALL when CLIENT_ACCEPTED { discard }',
<         'exit',
---
>         'create ltm rule /' + PART + '/' + PREFIX_NAME + '_DISCARD_ALL',
>         '   --> Copy and Paste the following between pre-included curly brackets <--',
>         'when CLIENT_ACCEPTED { discard }\n',
599a618,621
>     if user_args['ssl_public_ip'] is None:
>         user_args['ssl_public_ip'] = inventory_json['all']['vars']['external_lb_vip_address']
> 
>

**Comment by Mudpuppy12** [comment]
<code>
denn8098@MN80JQG3QN:/tmp$ diff f5-config.py ~/PycharmProjects/rpc_lem/lem-aide/user-tools/f5_config.py 
19c19
< import os
---
> import os,sys
22a23,38
> if os.environ.has_key('LAB_DATA_DIR'):
>    # If we have LAD_DATA_DIR set, we are operating inside a lab environment
>    LAB_DATA_DIR = os.environ['LAB_DATA_DIR']
> else:
>        print "Incorrect LEM install missing LAB_DATA_DIR"
>        sys.exit(1)
> 
> 
> if os.environ.has_key('LAB_NAME'):
>     LAB_NAME = os.environ['LAB_NAME'].lower()
> else:
>       print "Incorrect LEM install missing LAB_NAME"
>       sys.exit(1)
> 
> PART = LAB_NAME.upper()
> PREFIX_NAME = LAB_NAME.upper()
24,25d39
< PART = 'RPC'
< PREFIX_NAME = 'RPC'
29a44
> 
41a57,59
> 
> ## TODO: Configure monitoring user
> 
54c72
<     r' destination *:80 recv "302 Found" send "HEAD / HTTP/1.1\r\nHost:'
---
>     r' destination *:80 recv "200 OK" send "HEAD / HTTP/1.1\r\nHost:'
60c78
<     r' https destination *:443 recv "302 FOUND" send "HEAD / HTTP/1.1\r\nHost:'
---
>     r' https destination *:443 recv "200 OK" send "HEAD / HTTP/1.1\r\nHost:'
255c273
<     'nova_console': {
---
>     'nova_spice_console': {
454c472
<         default=None
---
>         default=LAB_NAME.lower() + ".rpc.rackspace.com"
544,546c562,564
<         'run util bash',
<         'tmsh create ltm rule /' + PART + '/' + PREFIX_NAME + '_DISCARD_ALL when CLIENT_ACCEPTED { discard }',
<         'exit',
---
>         'create ltm rule /' + PART + '/' + PREFIX_NAME + '_DISCARD_ALL',
>         '   --> Copy and Paste the following between pre-included curly brackets <--',
>         'when CLIENT_ACCEPTED { discard }\n',
599a618,621
>     if user_args['ssl_public_ip'] is None:
>         user_args['ssl_public_ip'] = inventory_json['all']['vars']['external_lb_vip_address']
> 
> 
</code>

**Comment by Mudpuppy12** [comment]
<pre>
denn8098@MN80JQG3QN:/tmp$ diff f5-config.py ~/PycharmProjects/rpc_lem/lem-aide/user-tools/f5_config.py 
19c19
< import os
---
> import os,sys
22a23,38
> if os.environ.has_key('LAB_DATA_DIR'):
>    # If we have LAD_DATA_DIR set, we are operating inside a lab environment
>    LAB_DATA_DIR = os.environ['LAB_DATA_DIR']
> else:
>        print "Incorrect LEM install missing LAB_DATA_DIR"
>        sys.exit(1)
> 
> 
> if os.environ.has_key('LAB_NAME'):
>     LAB_NAME = os.environ['LAB_NAME'].lower()
> else:
>       print "Incorrect LEM install missing LAB_NAME"
>       sys.exit(1)
> 
> PART = LAB_NAME.upper()
> PREFIX_NAME = LAB_NAME.upper()
24,25d39
< PART = 'RPC'
< PREFIX_NAME = 'RPC'
29a44
> 
41a57,59
> 
> ## TODO: Configure monitoring user
> 
54c72
<     r' destination *:80 recv "302 Found" send "HEAD / HTTP/1.1\r\nHost:'
---
>     r' destination *:80 recv "200 OK" send "HEAD / HTTP/1.1\r\nHost:'
60c78
<     r' https destination *:443 recv "302 FOUND" send "HEAD / HTTP/1.1\r\nHost:'
---
>     r' https destination *:443 recv "200 OK" send "HEAD / HTTP/1.1\r\nHost:'
255c273
<     'nova_console': {
---
>     'nova_spice_console': {
454c472
<         default=None
---
>         default=LAB_NAME.lower() + ".rpc.rackspace.com"
544,546c562,564
<         'run util bash',
<         'tmsh create ltm rule /' + PART + '/' + PREFIX_NAME + '_DISCARD_ALL when CLIENT_ACCEPTED { discard }',
<         'exit',
---
>         'create ltm rule /' + PART + '/' + PREFIX_NAME + '_DISCARD_ALL',
>         '   --> Copy and Paste the following between pre-included curly brackets <--',
>         'when CLIENT_ACCEPTED { discard }\n',
599a618,621
>     if user_args['ssl_public_ip'] is None:
>         user_args['ssl_public_ip'] = inventory_json['all']['vars']['external_lb_vip_address']
> 
> 
</pre>

**Comment by Mudpuppy12** [comment]
https://gist.github.com/Mudpuppy12/6295b72d268d8dc26bd81eec8af6031b

**Comment by git-harry** [comment]
@prometheanfire please don't remove these labels, they are required to facilitate appropriate tracking on the board.

**Comment by prometheanfire** [comment]
@git-harry moving it to doing removed it I think...  I didn't do it any other way :(

**Comment by git-harry** [comment]
@prometheanfire - that's weird. Okay, we'll have to keep an eye on that otherwise the board is going to turn into a complete mess.

/cc @b3rnard0

**Comment by sigmavirus24** [comment]
So, what are the actual work items here? How can we move this forward?

**Comment by prometheanfire** [comment]
@Mudpuppy12 when do you use your script?  All LEM / LAB builds?

**Comment by prometheanfire** [comment]
here's a diff of @Mudpuppy12's diff to be flake8 (with max line length of 120) passing and should also by py3 compat (print and itteritems).

https://gist.github.com/prometheanfire/1f526777ed8bfb1dc9ea8107ff82cb5a

**Comment by prometheanfire** [comment]
here's the diff of master to my changes https://gist.github.com/prometheanfire/db7a7fa9fcc06ca4edb18df4a30cdb3c

**Comment by prometheanfire** [comment]
@Mudpuppy12 here is my diff that allows the script to be used in both LEM and PROD.  I wasn't sure about lines 138-147, so I left those in for now (I don't know f5 syntaxy things).

https://gist.github.com/prometheanfire/8731bdbcdb07caecc6cb0116a5eb20bb

**Comment by prometheanfire** [comment]
also, for others, this is what I want to be merging into master, https://gist.github.com/prometheanfire/8731bdbcdb07caecc6cb0116a5eb20bb

**Comment by prometheanfire** [comment]
update

I have a lab checked out, had to reboot all nodes so the br-mgnt came up, but can't access the last two storage nodes to reboot, have to wait for @Mudpuppy12, also this'd be nice to be done at the end of rekicks :D

I've also generated the script but cannot do anything with it as @Mudpuppy12 is out of action right now.

because of the aforementioned issues I'm marking this as blocked

**Comment by mancdaz** [comment]
@prometheanfire are there any updates on this issue at all? I know @andymcc helped you past the downed nodes issue you encountered last week. Did you manage to complete the testing?

**Comment by mattt416** [comment]
Hi @mancdaz, I've posted a details updated in https://github.com/rcbops/rpc-openstack/pull/1295 regarding SYD-SYD2-2.  I think most issues have been unblocked and @prometheanfire can now continue testing the F5 changes he made.

**Comment by mancdaz** [comment]
Thanks @mattt416

@prometheanfire can you please finish up the testing here so we can get this issue closed off. 

Please also make sure you update the card as you progress so that this can be picked up from where you leave off if necessary.

**Comment by prometheanfire** [comment]
I just saw this, so no update I'll VPN in tomorrow morning and continue though.
